File: frontend/src/types/dto/entity/GameDto.ts
import type { GameStateDto } from "./GameStateDto"
import type { LobbyDto } from "./LobbyDto"
import type { SudokuPuzzleDto } from "./SudokuPuzzleDto"

export type GameDto = {
    id: number,
    lobby: LobbyDto,
    sudokuPuzzle: SudokuPuzzleDto,
    gameStates: GameStateDto[]
}

File: backend/src/main/java/com/github/ryand6/sudokuweb/dto/entity/GameDto.java
package com.github.ryand6.sudokuweb.dto.entity;

import lombok.*;

import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class GameDto {

    private Long id;

    private LobbyDto lobby;

    private SudokuPuzzleDto sudokuPuzzle;

    private Set<GameStateDto> gameStates;

}
